lib/gooddata/lcm/helpers/check_helper.rb

```diff
@@ -16,18 +16,21 @@ def check_params(specification, params)
             type = specification[param_name][:type]
             if value.nil? || (value.is_a?(String) && value.empty?)
               if specification[param_name][:opts][:default]
-                params[param_name] = specification[param_name][:opts][:default]
+                if specification.select { |x| specification[x][:opts][:replacement] == param_name }.first.nil?
```

Comment:
Refers to 'specification' more than self (maybe move it to another class?) (FeatureEnvy)